server/tasks/process_file.py

```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
     asyncio.run(agent.environment.finalize())
 
 
-@app.task(name="process_file_streaming", track_started=True, bind=True)
-def process_file_streaming(self, serialized_agent: bytes):
-    # Load the agent
-    agent = pickle.loads(serialized_agent)
+@app.task(
+    name="streaming_parent_task", track_started=True, bind=True, serializer="pickle"
+)
+def streaming_parent_task(
+    self, agent: Agent, result_handler: ResultHandler, batch_size: int = 2
+):
+    # Parent job ID is used for input/output topic names
+    parent_job_id = self.request.id
+
+    task = process_file_streaming
```

Comment:
nit: I know we just overwrite this later, but can we just name this `inference_task` or something? Also, in general, I think "inference" is more descriptive here than "input". If it seems clear to you too, imo we should just rename these vars to make the flow more clear.

Reply:
If you mean the left hand side `task` then yes I understand, agreed